sgkit_vcf/utils.py

```diff
@@ -23,9 +28,19 @@ def chunks(iterator: Iterator[T], n: int) -> Iterator[Iterator[T]]:
         yield itertools.chain([first], rest_of_chunk)  # concatenate the first item back
 
 
-def get_file_length(path: Path) -> int:
+def get_file_length(
+    path: PathType, storage_options: Optional[Dict[str, str]] = None
+) -> int:
     """Get the length of a file in bytes."""
-    return path.stat().st_size
+    if isinstance(path, Path):
+        return path.stat().st_size
+    else:
+        storage_options = storage_options or {}
+        fs = fsspec.open(path, **storage_options).fs
```

Comment:
Should this be done in a context manager?

Reply:
Done in https://github.com/pystatgen/sgkit-vcf/pull/6/commits/3eb0acfeb17919442f2aaf8ed60a431340f164a3